plugins/modules/elastic_agentpolicy_info.py

```diff
@@ -59,9 +59,12 @@ def main():
     else:
       agent_policy_object = kibana.get_agent_policy_byid(agent_policy_id)
       
-    results['agent_policy_status'] = "Getting Agent Policy"
-    results['agent_policy_object'] = agent_policy_object
-    
+    if agent_policy_object:
+      results['agent_policy_status'] = "Agent Policy Found"
+      results['agent_policy_object'] = agent_policy_object
```

Comment:
We should add `results['agent_policy_object'] = None` if no agent policy is found so our results are consistent.